cmd/snap-mgmt/snap-mgmt.sh.in

```diff
@@ -119,6 +119,28 @@ purge() {
         rm -f "/etc/systemd/system/multi-user.target.wants/$unit"
     done
 
+    # TODO: ask snapd to ask each snapd.session-agent.service to stop snaps
+    # user-session services and stop itself.
+
+    user_units=$(systemctl --user --global list-unit-files --no-legend --full)
+    # user services from snaps
+    user_services=$(echo "$user_units" | grep '^snap\..*\.service' | cut -f1 -d ' ')
+    for unit in $user_services; do
+        # ensure its really a systemd user service
+        if ! grep -q 'X-Snappy=yes' "/etc/systemd/user/$unit"; then
+            echo "Skipping non-snapd systemd user unit $unit"
+            continue
+        fi
+
+        echo "Removing $unit"
+        rm -f "/etc/systemd/user/$unit"
```

Comment:
We support sockets too, so we're leaving the socket units behind effectively.

Reply:
Ah, indeed. I'll look at that.

Reply:
Fixed

Reply:
Fixed